src/mlpack/methods/sparse_svm/sparse_svm.cpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file sparse_svm.cpp
+ * @author Ayush Chamoli
+ *
+ * Implementation of Sparse SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_SPARSE_SVM_SPARSE_SVM_CPP
+#define MLPACK_METHODS_SPARSE_SVM_SPARSE_SVM_CPP
+
+// In case it hasn't been included yet.
+#include "sparse_svm.hpp"
+
+namespace mlpack {
+namespace svm {
+
+template <typename OptimizerType>
+SparseSVM::SparseSVM(
```

Comment:
Since this is a templated function, we'd actually need to put it in a header file.  You could move it (and the other templated methods in this file) to `sparse_svm_impl.hpp`, which you can then include from `sparse_svm.hpp`.